src/components/PersonLink.tsx

```diff
@@ -0,0 +1,98 @@
+import classNames from 'classnames';
+import { Person } from '../types';
+import { NavLink, redirect, useLocation } from 'react-router-dom';
+
+type Props = {
+  people: Person;
+  peopleList: Person[];
+};
+
+export const PersonLink: React.FC<Props> = ({ people, peopleList }) => {
+  const location = useLocation();
+
+  const isNameInTable = peopleList
+    .map(person => person.name)
+    .includes(people.name);
+
+  const isFatherNameInTable = people.fatherName
+    ? peopleList.map(person => person.name).includes(people.fatherName)
+    : false;
+
+  const isMotherNameInTable = people.motherName
+    ? peopleList.map(person => person.name).includes(people.motherName)
+    : false;
+
+  const isFemale = (name: string) =>
+    peopleList.map(person =>
+      person.name === name && person.sex === 'f' ? true : false,
+    );
```

Comment:
The `isFemale` function returns an array of booleans instead of a single boolean value. This might cause unexpected behavior when used in className logic. Consider revising the function to return a single boolean value indicating if the name belongs to a female.